core-java-8/src/main/java/com/baeldung/findanelement/FindElementInAList.java

```diff
@@ -0,0 +1,72 @@
+package com.baeldung.findanelement;
+
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Optional;
+import org.apache.commons.collections4.IterableUtils;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+
+public class FindElementInAList<T> {
+
+    public T findUsingIndexOf(T element, List<T> listOfIntegers) {
+        int index = listOfIntegers.indexOf(element);
+        if (index >= 0) {
+            return element;
+        }
+        return null;
+    }
+
+    public T findUsingListIterator(T element, List<T> listOfIntegers) {
+        ListIterator<T> listIterator = listOfIntegers.listIterator();
+        while (listIterator.hasNext()) {
+            T elementFromList = listIterator.next();
+            if (elementFromList.equals(element)) {
+                return element;
+            }
+        }
+        return null;
+    }
+
+    public T findUsingEnhancedForLoop(T element, List<T> listOfIntegers) {
+        for (T elementFromList : listOfIntegers) {
+            if (element.equals(elementFromList)) {
+                return elementFromList;
+            }
+        }
+        return null;
+    }
+
+    public T findUsingStream(T element, List<T> listOfIntegers) {
+        Optional<T> foundElement = listOfIntegers.stream()
+            .filter(integer -> integer.equals(element))
+            .findFirst();
+        return foundElement.isPresent() ? foundElement.get() : null;
```

Comment:
I kind of like the look of
```
T foundElement = listOfInteger.stream()
  .filter(integer -> integer.equals(element))
  .findFirst()
  .orElse(null);
```
Since it makes it match Guava. Thoughts?

Reply:
Yes, that's same.
Updating to orElse ,this would be helpful when use compares between different approaches.